test/RawRabbit.IntegrationTests/SimpleUse/PublishAndSubscribeTests.cs

```diff
@@ -15,399 +15,509 @@
-		}
-
-		[Fact]
-		public async Task Should_Be_Able_To_Subscibe_To_Pure_Json_Message()
-		{
-			var conventions = new NamingConventions();
-			using (var client = TestClientFactory.CreateNormal(ioc => ioc.AddSingleton<INamingConventions>(c => conventions)))
-			{
-				/* Setup */
-				var tcs = new TaskCompletionSource<BasicMessage>();
-				var subscription = client.SubscribeAsync<BasicMessage>((message, context) =>
-				{
-					tcs.TrySetResult(message);
-					return Task.FromResult(true);
-				});
-				var uniqueValue = Guid.NewGuid().ToString();
-				var jsonMsg = JsonConvert.SerializeObject(new BasicMessage { Prop = uniqueValue });
-
-				/* Test */
-				TestChannel.BasicPublish(
-					conventions.ExchangeNamingConvention(typeof(BasicMessage)),
-					conventions.QueueNamingConvention(typeof(BasicMessage)),
-					true,
-					null,
-					Encoding.UTF8.GetBytes(jsonMsg));
-				await tcs.Task;
-
-				/* Assert */
-				Assert.Equal(uniqueValue, tcs.Task.Result.Prop);
-			}
-		}
-
-		[Fact]
-		public async void Should_Be_Able_To_Publish_Dynamic_Objects()
-		{
-			using (var client = TestClientFactory.CreateNormal())
-			{
-				/* Setup */
-				var tcs = new TaskCompletionSource<DynamicMessage>();
-				client.SubscribeAsync<DynamicMessage>((message, context) =>
-				{
-					tcs.TrySetResult(message);
-					return Task.FromResult(true);
-				});
-
-				/* Test */
-				client.PublishAsync(new DynamicMessage { Body = new { IsDynamic = true } });
-				await tcs.Task;
-
-				/* Assert */
-				Assert.True(tcs.Task.Result.Body.IsDynamic);
-			}
-		}
-
-		[Fact]
-		public async Task Should_Be_Able_To_Publish_Message_After_Failed_Publish()
-		{
-			using(var firstClient = TestClientFactory.CreateNormal())
-			using (var secondClient = TestClientFactory.CreateNormal())
-			{
-				/* Setup */
-				var tcs = new TaskCompletionSource<bool>();
-				firstClient.SubscribeAsync<SimpleMessage>((message, context) =>
-				{
-					tcs.TrySetResult(true);
-					return Task.FromResult(true);
-				});
-
-				/* Test */
-				try
-				{
-					await
-						secondClient.PublishAsync(new SimpleMessage(),
-							configuration: cfg => cfg.WithExchange(e => e.WithType(ExchangeType.Direct)));
-				}
-				catch (Exception)
-				{
-					await Task.Delay(50);
-					Assert.False(tcs.Task.IsCompleted);
-				}
-				secondClient.PublishAsync(new SimpleMessage());
-				await tcs.Task;
-				/* Assert */
-				Assert.True(true);
-			}
-		}
-
-		[Fact]
-		public async Task Should_Run_Basic_Return_When_The_Manatory_Set_After_Failed_Publish()
-		{
-			/* Setup */
-			using (var client = TestClientFactory.CreateNormal())
-			{
-				var tcs = new TaskCompletionSource<bool>();
-				/* Test */
-				await client.PublishAsync(new SimpleMessage(),
-					configuration: cfg => cfg
-						.WithMandatoryDelivery((obj, evt) =>
-						{
-							tcs.SetResult(true);
-						}));
-
-				/* Assert */
-				Assert.True(tcs.Task.Result);
-			}
-		}
-	}
+    using Context;
+    using RabbitMQ.Client.Exceptions;
```

Comment:
I probably hit resharper's cleaning func unknowingly  :smile: 

but afaik it's better to have usings inside the namespace to prevent conflicts